corehq/apps/app_manager/views/utils.py

```diff
@@ -623,10 +657,37 @@ def _is_duplicate_endpoint_id(new_id, old_id, app):
         all_endpoint_ids.append(module.session_endpoint_id)
         for form in module.get_suite_forms():
             all_endpoint_ids.append(form.session_endpoint_id)
+        if module.module_type == "shadow":
+            for m in module.form_session_endpoint_mapping:
+                all_endpoint_ids.append(m.session_endpoint_id)
 
     return new_id in all_endpoint_ids
 
 
+def _duplicate_endpoint_ids(new_session_endpoint_id, new_form_session_endpoint_ids, module_id, app):
+    all_endpoint_ids = [new_session_endpoint_id] + new_form_session_endpoint_ids
+
+    for module in app.modules:
+        if module.unique_id != module_id:
+            all_endpoint_ids.append(module.session_endpoint_id)
+            for form in module.get_suite_forms():
+                all_endpoint_ids.append(form.session_endpoint_id)
+            if module.module_type == "shadow":
+                for m in module.form_session_endpoint_mapping:
+                    all_endpoint_ids.append(m.session_endpoint_id)
+
+    seen = set()
+    duplicates = set()
+
+    for endpoint_id in all_endpoint_ids:
+        if endpoint_id in seen:
+            duplicates.add(endpoint_id)
+        else:
+            seen.add(endpoint_id)
```

Comment:
`collections.Counter` is great for this kind of thing:
```suggestion
    duplicates = [k for k, v in Counter(all_endpoint_ids).items() if v > 1]
```